ocaml/xapi/xapi_xenops.ml

```diff
@@ -1893,6 +1938,46 @@ let update_vgpu ~__context id =
   with e ->
     error "xenopsd event: Caught %s while updating VGPU" (string_of_exn e)
 
+let update_vusb ~__context (id: (string * string)) =
+  try
+    let open Vusb in
+    if Events_from_xenopsd.are_suppressed (fst id)
+    then debug "xenopsd event: ignoring event for VM (VM %s migrating away)" (fst id)
+    else
+      let vm = Db.VM.get_by_uuid ~__context ~uuid:(fst id) in
+      let localhost = Helpers.get_localhost ~__context in
+      if Db.VM.get_resident_on ~__context ~self:vm <> localhost
+      then debug "xenopsd event: ignoring event for VUSB (VM %s not resident)" (fst id)
+      else
+        let previous = Xenops_cache.find_vusb id in
+        let dbg = Context.string_of_task __context in
+        let module Client = (val make_client (queue_of_vm ~__context ~self:vm) : XENOPS) in
+        let info = try Some(Client.VUSB.stat dbg id) with _ -> None in
+        if Opt.map snd info = previous
+        then debug "xenopsd event: ignoring event for VUSB %s.%s: metadata has not changed" (fst id) (snd id)
+        else begin
+          let vusbs = Db.VM.get_VUSBs ~__context ~self:vm in
```

Comment:
See my comment above how this could be  simplified using `|>`. But you can keep this for now - I understand its like this somewhere else.

Reply:
updated.